triples-generator/src/main/scala/io/renku/triplesgenerator/events/categories/triplesgenerated/projectinfo/ProjectEventsFinder.scala

```diff
@@ -0,0 +1,100 @@
+package io.renku.triplesgenerator.events.categories.triplesgenerated.projectinfo
+
+import cats.data.EitherT
+import cats.effect.Async
+import cats.syntax.all._
+import eu.timepit.refined.auto._
+import io.circe.Decoder
+import io.circe.Decoder.decodeList
+import io.renku.graph.model.events.CommitId
+import io.renku.graph.model.{persons, projects}
+import io.renku.http.client.RestClient.ResponseMappingF
+import io.renku.http.client.{AccessToken, GitLabClient}
+import io.renku.triplesgenerator.events.categories.ProcessingRecoverableError
+import io.renku.triplesgenerator.events.categories.triplesgenerated.RecoverableErrorsRecovery
+import org.http4s.EntityDecoder
+import org.http4s.Method.GET
+import org.http4s.dsl.io.{NotFound, Ok}
+import org.http4s.implicits.http4sLiteralsSyntax
+import org.typelevel.ci._
+import org.typelevel.log4cats.Logger
+
+private trait ProjectEventsFinder[F[_]] {
+  def find(project:     Project, page: Int)(implicit
+      maybeAccessToken: Option[AccessToken]
+  ): EitherT[F, ProcessingRecoverableError, (List[PushEvent], PagingInfo)]
+}
+
+private class ProjectEventsFinderImpl[F[_]: Async: Logger](
+    gitLabClient:     GitLabClient[F],
+    recoveryStrategy: RecoverableErrorsRecovery = RecoverableErrorsRecovery
+) extends ProjectEventsFinder[F] {
+
+  override def find(project: Project, page: Int)(implicit
+      maybeAccessToken:      Option[AccessToken]
+  ): EitherT[F, ProcessingRecoverableError, (List[PushEvent], PagingInfo)] =
+    EitherT {
+      gitLabClient
+        .send(
+          GET,
+          uri"projects" / project.id.show / "events" withQueryParams Map("action" -> "pushed", "page" -> page.toString),
+          "events"
```

Comment:
Maybe we can use name to be a bit better distinguishable on the Grafana panel, e.g. `project-events` or smth better.

Reply:
I agree. I think the GitLab API docs call it simply "events" but I'll change it.